capstone/src/main/java/com/google/step/servlets/LoginServlet.java

```diff
@@ -0,0 +1,103 @@
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that will handle login information.
+ */
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    UserService userService = UserServiceFactory.getUserService();
+    Map<String, Object> login = new HashMap<>();
+
+    if (userService.isUserLoggedIn()) {
+      String useremail = userService.getCurrentUser().getEmail();
+      String type = checkIfRestaurantOrUser(useremail);
+
+      // If user is loggedin but not in our system.
+      if (type == null) {
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        return;
+      }
+
+      String urltoRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urltoRedirectToAfterUserLogsOut);
+
+      login.put("Loggedin", true);
+      login.put("Type", type);
+      login.put("Email", useremail);
+      login.put("logOutUrl", logoutUrl);
+    } else {
+      String urltoRedirectToAfterUserLogIn = "/";
+      String urltoRedirectAfterUserSignUp = "/user-signup.html";
+      String urltoRedirectAfterRestaurantSignUp = "/restaurant-signup.html";
+
+      String loginUrl = userService.createLoginURL(urltoRedirectToAfterUserLogIn);
+      String userSignUpUrl = userService.createLoginURL(urltoRedirectAfterUserSignUp);
+      String restaurantSignUpUrl = userService.createLoginURL(urltoRedirectAfterRestaurantSignUp);
+
+      login.put("Loggedin", false);
+      login.put("loginUrl", loginUrl);
+      login.put("userSignUpUrl", userSignUpUrl);
+      login.put("restaurantSignUpUrl", restaurantSignUpUrl);
+    }
+
+    String json = convertToJsonUsingGson(login);
+    response.getWriter().println(json);
+  }
+
+  // TODO: Change to have object return and return User or Restaurant Object
+  /**
+   * Checks if the current email is from a user or a restaurant.
+   * @param email of the current user.
+   * @return String containing the words User or Restaurant.
+   */
+  private static String checkIfRestaurantOrUser(String email) {
```

Comment:
Above that i have a TODO which in the future when we have developed the restaurant and user classes and have info, we change the method so it returns a class instead of a string. This is just a place holder for now.

Reply:
Oh, okay. I see. Do you plan to use the data in the restaurant/user object? If the information passed back is going to be the same, it might make more sense to pass around less information, in which case, I think the enum should do the trick.

If you need the returned object for something, perhaps it would make sense to split up the method into two separate methods, one returning a User object, and another returning a restaurant object? If there are going to be two different types returned from the same method, each in a different branch of the method, it might make sense to just separate the functionality completely.

Reply:
I was thinking of it more in terms of having return the object in order to have the information available to the user should he want to view his information. Rather than just using it for my method.

Reply:
Oh, so the information being returned by the servlet could possibly be changed from what it is right now? In that case, should the functionality in that method be broken up into two methods, perhaps a getRestaurantUser method and a getUser method? That way, it is very clear what is being returned from each method (User, RestaurantUser), and it better follows the Single-Responsibility Principle. Perhaps for now, because these objects are not being returned, you could have the two methods return a boolean?

Reply:
Yeah that would make sense. I will work on that then.

Reply:
Alright done